### PR TITLE
[AIDAPP-309]: After changing an article from public to internal, the article is still available via the link on the portal

### PR DESCRIPTION
**Pull request by ketan-canyon**
### Ticket(s) or GitHub Issue

- https://canyongbs.atlassian.net/browse/AIDAPP-309

### Technical Description

After changing an article from public to internal, the article is still available via the link on the portal

### Any deployment steps required?

No

### Are any Feature Flags and/or Data Migrations that can eventually be removed Added?

No

_______________________________________________

### CODE REVIEW

File: app-modules/portal/src/DataTransferObjects/KnowledgeBaseArticleData.php [hunk 1]
```diff
@@ -46,6 +46,6 @@ public function __construct(
         public string $name,
         public ?string $lastUpdated,
         public ?string $content,
-        public ?array $tags,
+        public ?array $tags
```
Comment (Orrison) [inline app-modules/portal/src/DataTransferObjects/KnowledgeBaseArticleData.php:49]: It is prefered to have trailing commas, please restore this.

File: app-modules/portal/src/Http/Controllers/KnowledgeManagementPortal/KnowledgeManagementPortalArticleController.php [hunk 2]
```diff
@@ -49,28 +49,32 @@ public function show(KnowledgeBaseCategory $category, KnowledgeBaseItem $article
...
+                            'name',
+                        ])
+                        ->get()
+                        ->toArray(),
+                ]),
+                'portal_view_count' => $article->portal_view_count,
+            ]);
+        }
+
+        return response()->json([]);
```
Comment (Orrison) [inline app-modules/portal/src/Http/Controllers/KnowledgeManagementPortal/KnowledgeManagementPortalArticleController.php:78]: You should refactor this to a negative check and an early return. Nesting the happy path can lead to complex and hard to read code:

```suggestion
        if (! $article->public) {
            return response()->json([], 401);
        }

        return response()->json([
                'category' => KnowledgeBaseCategoryData::from([
                    'id' => $category->getKey(),
                    'name' => $category->name,
                    'description' => $category->description,
                ]),
                'article' => KnowledgeBaseArticleData::from([
                    'id' => $article->getKey(),
                    'categoryId' => $article->category_id,
                    'name' => $article->title,
                    'lastUpdated' => $article->updated_at->format('M d Y, h:m a'),
                    'content' => tiptap_converter()->record($article, attribute: 'article_details')->asHTML($article->article_details),
                    'tags' => $article->tags()
                        ->orderBy('name')
                        ->select([
                            'id',
                            'name',
                        ])
                        ->get()
                        ->toArray(),
                ]),
                'portal_view_count' => $article->portal_view_count,
            ]);
```

**Review by Orrison — CHANGES_REQUESTED** [review]
Some comments. There are also some open merge conflicts

File: app-modules/portal/src/DataTransferObjects/KnowledgeBaseArticleData.php (hunk 1 above)
Comment (ketan-canyon) [reply app-modules/portal/src/DataTransferObjects/KnowledgeBaseArticleData.php]: Done

File: app-modules/portal/src/Http/Controllers/KnowledgeManagementPortal/KnowledgeManagementPortalArticleController.php (hunk 2 above)
Comment (ketan-canyon) [reply app-modules/portal/src/Http/Controllers/KnowledgeManagementPortal/KnowledgeManagementPortalArticleController.php]: Done

**Review by Orrison — APPROVED** [review]

**Pull request merged**